secure_fw/partitions/internal_trusted_storage/tfm_internal_trusted_storage.c

```diff
@@ -272,29 +300,50 @@ psa_status_t tfm_its_set(int32_t client_id,
     }
 
     offset = 0;
-    flags = (uint32_t)create_flags |
-            ITS_FLASH_FS_FLAG_CREATE | ITS_FLASH_FS_FLAG_TRUNCATE;
+    /* Populate the file info for the new file */
+    g_file_info.size_max = data_length;
+    g_file_info.flags = (uint32_t)create_flags |
+                        ITS_FLASH_FS_FLAG_CREATE | ITS_FLASH_FS_FLAG_TRUNCATE;
+
 
     /* Iteratively read data from the caller and write it to the filesystem, in
      * chunks no larger than the size of the asset_data buffer.
      */
     do {
-        /* Write as much of the data as will fit in the asset_data buffer */
         write_size = ITS_UTILS_MIN(data_length, sizeof(asset_data));
 
         /* Read asset data from the caller */
         (void)its_req_mngr_read(asset_data, write_size);
 
+#ifdef TFM_ITS_ENCRYPT
+    if (client_id != TFM_SP_PS) {
+        status = tfm_its_crypt_file(&g_file_info,
```

Comment:
This happens in a do-while loop. What happens if there are multiple iterations? It seems like the tag is overwritten each time, though I'm not entirely sure what is happening.

Reply:
When encryption is used we enforce the data_length to be <= to asset data buffer. This is done here:
https://github.com/nrfconnect/sdk-trusted-firmware-m/blob/b23ce979bdbac6424c51a233ded800f234039776/secure_fw/partitions/internal_trusted_storage/tfm_internal_trusted_storage.c#L264

I added a comment here to clarify this more.